tern/analyze/docker/container.py

```diff
@@ -96,6 +96,24 @@ def pull_image(image_tag_string):
         return False
 
 
+def get_image(image_tag_string):
+    '''Try to get an image from Dockerhub. If image is not valid
+    or unavailable return blank string.
+    image_tag_string: can be in image:tag or image@digesttype:digest format'''
+    check_docker_setup()
+    # Can only docker 'get' a local image. If not available, pull it first
+    if not check_image(image_tag_string):
+        if not pull_image(image_tag_string):
+            return ''
+    return client.images.get(image_tag_string)
```

Comment:
I was actually thinking you could modify the `check_image` function above to return the docker_image object instead of returning `True`. If no object is returned then return `None`. Same thing with the `pull_image` function (although I haven't tested whether it will return the same digest).
Also, this function would be useful in the `setup` function in `report.py`. Alternatively, you could also assume that the check for the image is already done and it is available (or the script has errored out).